test/pytest/test_example_stateful_sequence_continuous_batching_http.py

```diff
@@ -0,0 +1,363 @@
+        if sequence_id == "seq_0":
+            idx = 2 * idx
+        elif sequence_id == "seq_1":
+            idx = 2 * idx + 1
+        if start is True:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                data=str(idx + 1).encode(),
+            )
+            s_id = response.headers.get("ts_request_sequence_id")
+            if sequence_id == "seq_0":
+                headers_seq_0 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            elif sequence_id == "seq_1":
+                headers_seq_1 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            start = False
+        else:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                headers=headers_seq_0 if sequence_id == "seq_0" else headers_seq_1,
+                data=str(idx + 1).encode(),
+            )
+        prediction.append(response.text)
+
+    print(f"infer_stateful prediction={str(' '.join(prediction))}")
+    assert str(" ".join(prediction)) == expected
+
+
+def __infer_stateful_end(model_name, sequence_id, expected):
+    prediction = []
+    start = True
+    end = False
+    for idx in range(5):
+        if idx == 4:
+            end = True
+        if sequence_id == "seq_0":
+            idx = 2 * idx
+        elif sequence_id == "seq_1":
+            idx = 2 * idx + 1
+        if end is True:
+            idx = -1
+
+        if start is True:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                data=str(idx + 1).encode(),
+            )
+            s_id = response.headers.get("ts_request_sequence_id")
+            if sequence_id == "seq_0":
+                headers_seq_0 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            elif sequence_id == "seq_1":
+                headers_seq_1 = {
+                    "ts_request_sequence_id": s_id,
+                }
+            start = False
+        else:
+            response = requests.post(
+                url=f"http://localhost:8080/predictions/{model_name}",
+                headers=headers_seq_0 if sequence_id == "seq_0" else headers_seq_1,
+                data=str(idx + 1).encode(),
+            )
+        prediction.append(response.text)
+
+    print(f"infer_stateful_end prediction={str(' '.join(prediction))}")
+    assert str(" ".join(prediction)) == expected
+
+
+def __infer_stateful_cancel(model_name, sequence_id, expected):
+    prediction = []
+    start = True
+    cancel = False
+    for idx in range(5):
+        if idx == 2:
+            cancel = True
+        if sequence_id == "seq_0":
+            idx = 2 * idx
+        elif sequence_id == "seq_1":
+            idx = 2 * idx + 1
+
+        if cancel is True and sequence_id == "seq_0":
+            response = requests.post(
```

Comment:
This test does not cover the continuous batching part. We should add that in the test.

Reply:
it is complicate to match the final result b/c it is dynamically in continuous batching case. Here, the client side does not enable chunked encode, which mean skipping the rest of response, to get the fixed the result.

Reply:
You can just use sleep in the inference() call to make sure we get a cancellation request in after we get a certain response through the chunked response.